fix: Error in verifying the server X.509 certificate

splasky opened the pull request:
Fix error in verifying the server X.509 certificate.This
commit will set conn_http to use https as default. The default
certificate and private key is setting inside `pem/` for build system.
To replace the default certificate, add macro `PEM=/path/to/cert.pem`
for gnu make.

closes #6

jserv commented on connectivity/conn_http.c:
```diff
@@ -16,11 +16,11 @@
 
 #define MAX_PORT_LEN 5
 #define MAX_CONTENT_LENGTH_LEN 5
+#define CA_PATH "fullchain6.pem"
```
Prepare utilities to convert arbitrary file into C header to avoid file operations.

splasky replied:
The certificate will be put into secure storage. Should we convert it to C header here?
Related at #19 .

jserv replied:
If something is about to be preserved in secure storage, don't treat it as mixture.

Commit:
fix: Error in verifying the server X.509 certificate

Fix error in verifying the server X.509 certificate.This
commit will set conn_http to use https as default. The default
certificate and private key is setting inside `pem/` for build system.
To replace the default certificate, add macro `PEM=/path/to/cert.pem`
for gnu make.

closes #6

jserv commented on connectivity/ca_cert.h:
```diff
@@ -5,1148 +5,194 @@
-    "-----END CERTIFICATE-----\r\n"
-    " "};
\ No newline at end of file
+unsigned char ca_crt_pem[] = {
```
Should be `static`.

jserv commented:
Improve git commit messages.

jserv commented on Makefile:
```diff
@@ -53,11 +53,26 @@ utils:
 hal:
 	$(MAKE) -C $(HAL_PATH)
+cert: check_pem
+	@xxd -i $(PEM) > connectivity/ca_cert.inc
```
What is the default string for `$(PEM)`?

splasky replied:
`$(PEM)` is the certificate path. It should be provided as following command. `make PEM=ca.pem`. If the `$(PEM)` is not set, it will raise error.

jserv replied:
Let's set default and preserved path, otherwise it would raise problems with CI and general use.

jserv commented:
Don't merge into this repository. Move to TA instead.

splasky commented:
Merge into [feat(endpoint): Consolidate by merging Endpoint subproject
#570](https://github.com/DLTcollab/tangle-accelerator/pull/570)
